### ISSUE TITLE
arrow-parens: false positive for generic arrow function

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/typescript-eslint-parser#475]
**What version of TypeScript are you using?**
2.8.3

**What version of `typescript-eslint-parser` are you using?**
15.0.0

**What code were you trying to parse?**

```ts
// Unexpected error: arrow-parens
const bar = <T>(t: T) => {};
// No error as expected
const bar2 = function <T>(t: T) {};
```

**Comment by JamesHenry** [comment]
This makes sense, a custom rule will need to be created in the [plugin](https://github.com/nzakas/eslint-plugin-typescript) to handle this. It would be fantastic if one of the people affected by this issue could contribute to it.

**Comment by knpwrs** [comment]
For the time being I'm following this two-step plan:

## First: Disable eslint no-unused-vars for typescript files:

```json
  "overrides": [{
    "files": ["src/**/*.ts{,x}"],
    "parser": "typescript-eslint-parser",
    "rules": {
      "no-unused-vars": "off"
    },
  }],
```

## Second: Enable unused variable errors in TypeScript:

```json
    "noUnusedLocals": true,
```

**Comment by JamesHenry** [comment]
This issue has been migrated to the new project here: https://github.com/typescript-eslint/typescript-eslint/issues/14

Thanks!
